src/ResourceManagement/ResourceManager/Core/ExternalChildResourceCollection.cs

```diff
@@ -69,12 +69,15 @@ public Task<List<FluentModelTImpl>> CommitAndGetAllAsync(CancellationToken cance
                                 exceptions.Add(deleteTask.Exception);
                             }
                         }
+                        if (deleteTask.IsCanceled)
+                        {
+                            exceptions.Add(new TaskCanceledException());
```

Comment:
What's the difference between `IsFaulted` branch and `IsCanceled` branch?

Reply:
When getting 504 error, for `isFaulted`, the task would have taskCanceledException inside. But for `isCanceled`, the task would have exception as null.